boa/src/builtins/object/mod.rs

```diff
@@ -47,322 +46,85 @@ pub static PROTOTYPE: &str = "prototype";
 pub static INSTANCE_PROTOTYPE: &str = "__proto__";
 
 /// The internal representation of an JavaScript object.
-#[derive(Trace, Finalize, Clone)]
+#[derive(Debug, Trace, Finalize, Clone)]
 pub struct Object {
     /// The type of the object.
-    pub kind: ObjectKind,
+    pub data: ObjectData,
     /// Internal Slots
-    pub internal_slots: FxHashMap<String, Value>,
+    internal_slots: FxHashMap<String, Value>,
     /// Properties
-    pub properties: FxHashMap<String, Property>,
+    properties: FxHashMap<String, Property>,
     /// Symbol Properties
-    pub sym_properties: FxHashMap<i32, Property>,
+    symbol_properties: FxHashMap<u32, Property>,
     /// Some rust object that stores internal state
-    pub state: Option<InternalStateCell>,
-    /// Function
-    pub func: Option<Function>,
+    state: Option<InternalStateCell>,
 }
 
-impl Debug for Object {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        writeln!(f, "{{")?;
-        writeln!(f, "\tkind: {}", self.kind)?;
-        writeln!(f, "\tstate: {:?}", self.state)?;
-        writeln!(f, "\tfunc: {:?}", self.func)?;
-        writeln!(f, "\tproperties: {{")?;
-        for (key, _) in self.properties.iter() {
-            writeln!(f, "\t\t{}", key)?;
-        }
-        writeln!(f, "\t }}")?;
-        write!(f, "}}")
-    }
+/// Defines the different types of objects.
+#[derive(Debug, Trace, Finalize, Clone)]
+pub enum ObjectData {
+    Array,
```

Comment:
Just for context. why is it called `ObjectData`? 
Looks like it's still used for Type definition.

Reply:
Because it also holds data its not just type definitions, it's like `ValuData`, the doc comment is misleading,